src/ZFan.cpp

```diff
@@ -34,29 +34,28 @@ void ZFan::detach() {
 }
 
 void ZFan::setIndex(int8_t fanIndex) {
-  if(fanIndex != -1 && fanIndex < MAX_FANS) {
+  if(fanIndex >= 0 && fanIndex < MAX_FANS) {
     _fanIndex = fanIndex;
     fanInstances[_fanIndex] = this;
   }
 }
 
 void ZFan::setFanSpeed(uint8_t speed) {
-  _speed = speed;
```

Comment:
_speed is an internal member storing the current speed. Although it's not implemented yet, it's meant to retrieve the current speed setting using getFanSpeed().